syntax = "proto3";

// A backend message from the browser to the proxy.
message BackMsg {
  oneof type {
    // Set to true to ask the proxy to display the inline help.
    // bool help = 1;  DEPRECATED

    // Set to true to ask the proxy to upload the report.
    bool cloud_upload = 2;

    // Asks the proxy to run the script with this given command line.
    string rerun_script = 3;

    // DEPRECATED - Asks the proxy to run the script with this object
    // ReRun rerun = 4;

    // Requests that the report's @st_cache be cleared
    bool clear_cache = 5;

    // Requests that the Proxy's runOnSave behavior be set to the given value
    bool set_run_on_save = 6;
  }
}
